python/instrumentation/openinference-instrumentation-guardrails/src/openinference/instrumentation/guardrails/_wrap_guard_call.py

```diff
@@ -0,0 +1,264 @@
+import json
+from abc import ABC
+from typing import Any, Callable, Iterator, List, Mapping, Tuple
+from enum import Enum
+import opentelemetry.context as context_api
+from opentelemetry import trace as trace_api
+from opentelemetry.util.types import AttributeValue
+from openinference.instrumentation import get_attributes_from_context, safe_json_dumps
+from opentelemetry import context as otel_context
+from inspect import signature
+from openinference.semconv.trace import (
+    OpenInferenceSpanKindValues,
+    SpanAttributes
+)
+
+class SafeJSONEncoder(json.JSONEncoder):
+    """
+    Safely encodes non-JSON-serializable objects.
+    """
+
+    def default(self, o: Any) -> Any:
+        try:
+            return super().default(o)
+        except TypeError:
+            if hasattr(o, "dict") and callable(o.dict):  # pydantic v1 models, e.g., from Cohere
+                return o.dict()
+            return repr(o)
+
+def _flatten(mapping: Mapping[str, Any]) -> Iterator[Tuple[str, AttributeValue]]:
+    for key, value in mapping.items():
+        if value is None:
+            continue
+        if isinstance(value, Mapping):
+            for sub_key, sub_value in _flatten(value):
+                yield f"{key}.{sub_key}", sub_value
+        elif isinstance(value, List) and any(isinstance(item, Mapping) for item in value):
+            for index, sub_mapping in enumerate(value):
+                for sub_key, sub_value in _flatten(sub_mapping):
+                    yield f"{key}.{index}.{sub_key}", sub_value
+        else:
+            if isinstance(value, Enum):
+                value = value.value
+            yield key, value
+
+def _get_input_value(method: Callable[..., Any], *args: Any, **kwargs: Any) -> str:
+    """
+    Parses a method call's inputs into a JSON string. Ensures a consistent
+    output regardless of whether the those inputs are passed as positional or
+    keyword arguments.
+    """
+
+    # For typical class methods, the corresponding instance of inspect.Signature
+    # does not include the self parameter. However, the inspect.Signature
+    # instance for __call__ does include the self parameter.
+    method_signature = signature(method)
+    first_parameter_name = next(iter(method_signature.parameters), None)
+    signature_contains_self_parameter = first_parameter_name in ["self"]
```

Comment:
We don't actually have to worry about this because wrapt has taken care of it for us (by [shifting](https://github.com/GrahamDumpleton/wrapt/blob/5c0997c0b5be36da5f621c3473c95f9efe14891e/src/wrapt/wrappers.py#L449) the args). See below.

<img width="708" alt="Screenshot 2024-07-03 at 10 11 00 AM" src="https://github.com/Arize-ai/openinference/assets/80478925/cf26f29d-3373-4893-ad2b-5a08c155a4c3">